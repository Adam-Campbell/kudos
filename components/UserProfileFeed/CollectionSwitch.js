import { connect } from 'react-redux';
import SimpleArticleBlock from '../SimpleArticleBlock';
import SimpleCommentBlock from '../SimpleCommentBlock';
import SimpleKudosBlock from '../SimpleKudosBlock';
import SimpleHighlightBlock from '../SimpleHighlightBlock';

const UsersArticlesCollection = props => (
    <div>
        {props.article_ids.map((_id, index) => (
            <SimpleArticleBlock _id={_id} key={index} />
        ))}
    </div>
);

const UsersCommentsCollection = props => (
    <div>
        {props.comment_ids.map((_id, index) => (
            <SimpleCommentBlock _id={_id} key={index} />
        ))}
    </div>
);

const UsersKudosCollection = props => (
    <div>
        {props.kudos_ids.map((_id, index) => (
            <SimpleKudosBlock _id={_id} key={index} />
        ))}
    </div>
);

const UsersHighlightsCollection = props => (
    <div>
        {props.highlight_ids.map((_id, index) => (
            <SimpleHighlightBlock _id={_id} key={index} />
        ))}
    </div>
);

const CollectionSwitch = props => {
    const user = props.users[props.user_id];

    switch (props.filter) {
        case 'posts':
            return <UsersArticlesCollection article_ids={user.postIds} />

        case 'comments':
            return <UsersCommentsCollection comment_ids={user.commentIds} />

        case 'kudos':
            return <UsersKudosCollection kudos_ids={user.kudosIds} />

        case 'highlights':
            return <UsersHighlightsCollection highlight_ids={user.highlightIds} />

        default:
            return <UsersArticlesCollection article_ids={user.postIds} />
    }
}

const mapStateStateToProps = state => ({
    users: state.users.models
});

export default connect(mapStateStateToProps)(CollectionSwitch);
