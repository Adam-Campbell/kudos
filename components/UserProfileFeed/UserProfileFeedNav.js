import styled from 'styled-components';
import * as styleConstants from '../styleConstants';
import Link from 'next/link';

const Nav = styled.nav`

`;

const NavList = styled.ul`
    list-style: none;
    padding-left: 0;
`;

const NavItem = styled.li`
    display: inline-block;
`;

const NavAnchor = styled.a`
    display: block;
    font-family: ${styleConstants.fontSecondary};
    font-size: 16px;
    font-weight: ${props => props.active ? 400 : 300};
    color: ${styleConstants.colorBodyText};
    text-decoration: ${props => props.active ? 'underline': 'none'};
    padding: 8px 16px;
    &:hover {
        text-decoration: underline;
    }
`;

const UserProfileFeedNav = props => (
    <section>
        <Nav>
            <NavList>
                <NavItem>
                    <Link passHref as={`/user/${props.user_id}/posts`} href={`user?user=${props.user_id}&filter=posts`}>
                        <NavAnchor active={props.filter === 'posts' || props.filter === undefined}>Posts</NavAnchor>
                    </Link>
                </NavItem>
                <NavItem>
                    <Link passHref as={`/user/${props.user_id}/comments`} href={`user?user=${props.user_id}&filter=comments`}>
                        <NavAnchor active={props.filter === 'comments'}>Comments</NavAnchor>
                    </Link>
                </NavItem>
                <NavItem>
                    <Link passHref as={`/user/${props.user_id}/kudos`} href={`user?user=${props.user_id}&filter=kudos`}>
                        <NavAnchor active={props.filter === 'kudos'}>Kudos</NavAnchor>
                    </Link>
                </NavItem>
                <NavItem>
                    <Link passHref as={`/user/${props.user_id}/highlights`} href={`user?user=${props.user_id}&filter=highlights`}>
                        <NavAnchor active={props.filter === 'highlights'}>Highlights</NavAnchor>
                    </Link>
                </NavItem>
            </NavList>
        </Nav>
    </section>
);

export default UserProfileFeedNav;
