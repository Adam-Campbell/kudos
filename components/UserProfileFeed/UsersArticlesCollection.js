import PropTypes from 'prop-types';
import SimpleArticleBlock from '../SimpleArticleBlock';

const UsersArticlesCollection = props => (
    <div>
        {props.article_ids.map((article_id, index) => (
            <SimpleArticleBlock article_id={article_id} key={index} />
        ))}
    </div>
);

UsersArticlesCollection.propTypes = {
    article_ids: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default UsersArticlesCollection;