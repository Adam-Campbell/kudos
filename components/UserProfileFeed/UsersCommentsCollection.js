import PropTypes from 'prop-types';
import SimpleCommentBlock from '../SimpleCommentBlock';

const UsersCommentsCollection = props => (
    <div>
        {props.comment_ids.map((comment_id, index) => (
            <SimpleCommentBlock comment_id={comment_id} key={index} />
        ))}
    </div>
);

UsersCommentsCollection.propTypes = {
    comment_ids: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default UsersCommentsCollection;