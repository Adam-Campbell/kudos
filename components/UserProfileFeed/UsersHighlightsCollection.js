import PropTypes from 'prop-types';
import SimpleHighlightBlock from '../SimpleHighlightBlock';

const UsersHighlightsCollection = props => (
    <div>
        {props.highlight_ids.map((highlight_id, index) => (
            <SimpleHighlightBlock highlight_id={highlight_id} key={index} />
        ))}
    </div>
);

UsersHighlightsCollection.propTypes = {
    highlight_ids: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default UsersHighlightsCollection;