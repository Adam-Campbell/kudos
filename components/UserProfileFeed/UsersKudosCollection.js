import PropTypes from 'prop-types';
import SimpleKudosBlock from '../SimpleKudosBlock';

const UsersKudosCollection = props => (
    <div>
        {props.kudos_ids.map((article_id, index) => (
            <SimpleKudosBlock article_id={article_id} key={index} />
        ))}
    </div>
);

UsersKudosCollection.propTypes = {
    kudos_ids: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default UsersKudosCollection;