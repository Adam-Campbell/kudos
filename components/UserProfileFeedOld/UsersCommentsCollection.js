import Comment from '../Comment';

const UsersCommentsCollection = props => (
    <div>
        {props.commentIds.map((commentId, index) => (
            <Comment 
                key={index}
                _id={commentId}
            />
        ))}
    </div>
);

export default UsersCommentsCollection;