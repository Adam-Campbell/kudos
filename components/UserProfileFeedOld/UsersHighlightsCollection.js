import Highlight from '../Highlight';

const UsersHighlightsCollection = props => (
    <div>
        {props.highlightIds.map((highlightId, index) => (
            <Highlight 
                key={index}
                _id={highlightId}
            />
        ))}
    </div>
);

export default UsersHighlightsCollection;