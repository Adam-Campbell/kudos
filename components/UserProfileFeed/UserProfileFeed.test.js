import { shallow } from 'enzyme';
import { store } from '../../mockedStore';
import { UserProfileFeed } from './index';
import UserProfileFeedNav from './UserProfileFeedNav';
import { CollectionSwitch } from './CollectionSwitch';
import UsersArticlesCollection from './UsersArticlesCollection';
import UsersCommentsCollection from './UsersCommentsCollection';
import UsersHighlightsCollection from './UsersHighlightsCollection';
import UsersKudosCollection from './UsersKudosCollection';
const user_id = 'user_A';

describe('<UserProfileFeed/>', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <UserProfileFeed 
                user_id={user_id}
                filter={'posts'}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<UserProfileFeedNav/>', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <UserProfileFeedNav 
                user_id={user_id}
                filter={'posts'}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<CollectionSwitch/>', () => {
    test('renders correctly according the filter prop supplied to it', () => {
        const withPosts = shallow(
            <CollectionSwitch 
                user_id={user_id}
                users={store.users}
                filter={'posts'}
            />
        );
        expect(withPosts).toMatchSnapshot();
        expect(withPosts.find('UsersArticlesCollection')).toHaveLength(1);
        expect(withPosts.find('UsersCommentsCollection')).toHaveLength(0);
        const withComments = shallow(
            <CollectionSwitch 
                user_id={user_id}
                users={store.users}
                filter={'comments'}
            />
        );
        expect(withComments).toMatchSnapshot();
        expect(withComments.find('UsersCommentsCollection')).toHaveLength(1);
        expect(withComments.find('UsersArticlesCollection')).toHaveLength(0);
    });
});

describe('<UsersArticlesCollection', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <UsersArticlesCollection 
                article_ids={store.users.user_A.postIds}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<UsersCommentsCollection', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <UsersCommentsCollection 
                comment_ids={store.users.user_A.commentIds}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<UsersKudosCollection', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <UsersKudosCollection 
                kudos_ids={store.users.user_A.kudosIds}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<UsersHighlightsCollection', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <UsersHighlightsCollection 
                highlight_ids={store.users.user_A.highlightIds}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
