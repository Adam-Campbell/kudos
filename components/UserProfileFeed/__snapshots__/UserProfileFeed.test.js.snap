// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionSwitch/> renders correctly according the filter prop supplied to it 1`] = `
<UsersArticlesCollection
  article_ids={
    Array [
      "article_A",
      "article_C",
    ]
  }
/>
`;

exports[`<CollectionSwitch/> renders correctly according the filter prop supplied to it 2`] = `
<UsersCommentsCollection
  comment_ids={
    Array [
      "comment_A",
      "comment_B",
    ]
  }
/>
`;

exports[`<UserProfileFeed/> renders correctly 1`] = `
<section>
  <UserProfileFeedNav
    filter="posts"
    user_id="user_A"
  />
  <Connect(CollectionSwitch)
    filter="posts"
    user_id="user_A"
  />
</section>
`;

exports[`<UserProfileFeedNav/> renders correctly 1`] = `
<section>
  <Nav>
    <NavList>
      <NavItem>
        <Link
          as="/user/user_A/posts"
          href="user?user=user_A&filter=posts"
          passHref={true}
        >
          <NavAnchor
            active={true}
          >
            Posts
          </NavAnchor>
        </Link>
      </NavItem>
      <NavItem>
        <Link
          as="/user/user_A/comments"
          href="user?user=user_A&filter=comments"
          passHref={true}
        >
          <NavAnchor
            active={false}
          >
            Comments
          </NavAnchor>
        </Link>
      </NavItem>
      <NavItem>
        <Link
          as="/user/user_A/kudos"
          href="user?user=user_A&filter=kudos"
          passHref={true}
        >
          <NavAnchor
            active={false}
          >
            Kudos
          </NavAnchor>
        </Link>
      </NavItem>
      <NavItem>
        <Link
          as="/user/user_A/highlights"
          href="user?user=user_A&filter=highlights"
          passHref={true}
        >
          <NavAnchor
            active={false}
          >
            Highlights
          </NavAnchor>
        </Link>
      </NavItem>
    </NavList>
  </Nav>
</section>
`;

exports[`<UsersArticlesCollection renders correctly 1`] = `
<div>
  <Connect(SimpleArticleBlockContainer)
    article_id="article_A"
    key="0"
  />
  <Connect(SimpleArticleBlockContainer)
    article_id="article_C"
    key="1"
  />
</div>
`;

exports[`<UsersCommentsCollection renders correctly 1`] = `
<div>
  <Connect(SimpleCommentBlockContainer)
    comment_id="comment_A"
    key="0"
  />
  <Connect(SimpleCommentBlockContainer)
    comment_id="comment_B"
    key="1"
  />
</div>
`;

exports[`<UsersHighlightsCollection renders correctly 1`] = `
<div>
  <Connect(SimpleHighlightBlockContainer)
    highlight_id="article_A"
    key="0"
  />
</div>
`;

exports[`<UsersKudosCollection renders correctly 1`] = `
<div>
  <Connect(SimpleKudosBlockContainer)
    article_id="article_A"
    key="0"
  />
</div>
`;
